src/app/projects/add-edit-project/add-edit-project.component.ts

```diff
@@ -791,81 +850,46 @@ export class AddEditProjectComponent implements OnInit, AfterViewInit, OnDestroy
   }
 
   /**
-   * Publish the selected logos.
+   * Publish the selected logos or shapefiles. Logos and shapefiles are already saved to the DB
+   * when selected in the file picker.
    *
    * @returns {void}
    */
-  private publishSelectedLogos() {
+  private publishAttachedFiles() {
     const logoValues = this.getLogosFormValues();
-    const documentPublishRequests = logoValues.map(logo => {
-      return this.documentService.publish(logo.document);
+    const shapefileValues = this.getShapefilesFormValues();
+    // Filters down to the IDs of the documents/files to publish
+    const logoDocuments = logoValues.map(logo => logo.document);
+    const shapefileDocuments = shapefileValues.map(shapefile => shapefile.document);
+    const allDocuments = logoDocuments.concat(shapefileDocuments);
+    
+    const documentPublishRequests = allDocuments.map(doc => {
+      return this.documentService.publish(doc);
     });
+
     forkJoin(documentPublishRequests)
     .subscribe(
       aggregateResponse => {
         aggregateResponse.forEach((individualResponse: Document|HttpErrorResponse) => {
           if ("status" in individualResponse) {
-            // One or more of the responses is an error.
+            // If one or more of the responses is an error.
             if (500 === individualResponse.status || 400 === individualResponse.status) {
               console.error('Error publishing file', individualResponse);
-              alert('There was a problem publishing one or more of the selected logos.')
+              alert('There was a problem publishing one or more of the selected logos/shapefiles.')
               return;
             }
           }
         });
       },
       error => {
         console.error('Error publishing files', error);
-        alert('There was a problem publishing one or more of the selected logos.')
+        alert('There was a problem publishing one or more of the selected logos/shapefiles.')
         return;
       },
       () => {} // On finished.
     )
   }
 
-  /**
```

Comment:
Curious why we got rid of this code. Did you find a new way to dynamically change the text colour on top of the colour background?

Reply:
The ticket didn't call for any kind of hex value to be displayed prominently to the user when it came to the individual shapefile project filters so instead of hooking it up to the new system I got rid of it and opted for a simple colour display. The hex value is still present when they open the colour picker

Reply:
That's fair regarding the change in interface. Do you think this code has a home in the utilities section for possible dark/light mode parsing?

Reply:
If the code isn't actively being used anywhere then I can't think of a reason why we'd want to keep it